app/src/main/java/com/pusher/demo/features/marketplace/seller/SellerPresenter.kt

```diff
@@ -36,59 +40,68 @@ class SellerPresenter :  BasePresenter<SellerPresenter.View>(){
             })
     }
 
-    fun subscribeToRoom(room: Room) {
+    fun endSubscriptionToRoomUpdates() {
+        ChatkitManager.removeChatListener(chatListeners)
+    }
+
+    fun subscribeToRoomUpdates() {
 
         if (ChatkitManager.currentUser == null) {
             view?.onError("Current user was not found - have you signed in?")
             Log.e(ChatkitManager.LOG_TAG, "Current user was not found - have you signed in?")
             return
         }
 
-        //subscribe to the room
-        ChatkitManager.currentUser!!.subscribeToRoomMultipart(
-            roomId = room.id ,
-            listeners = RoomListeners(
-                onPresenceChange = { person ->
-                    if (person.id != ChatkitManager.currentUser!!.id) {
-                        view?.onMemberPresenceChanged(person)
-                    }
-                },
-                onMultipartMessage = {
-                    view?.onUnreadCountChanged(it.room)
-                }
-            ),
-            messageLimit = 0,
-            callback = { subscription ->
-                //success
-                getMembersForRoom(room)
+        chatListeners = ChatListeners(
+            onRoomUpdated = {
+                view?.onUnreadCountChanged(it)
+            },
+            onPresenceChanged = { user: User, newPresence: Presence, oldPresence: Presence ->
+                view?.onMemberPresenceChanged(user)
             }
         )
+
+        ChatkitManager.addChatListener(chatListeners)
+
+        ChatkitManager.getAllUsersInitialState(object: ChatkitManager.ChatManagerAllUsersListener{
+            override fun onUsers(users: List<User>) {
+                reconcileUsers(users)
+            }
+
+            override fun onError(error: String) {
+                //todo: handle this D:
+            }
+
+        })
+
     }
 
-    private fun getMembersForRoom(room: Room){
-        //get members for room
-        ChatkitManager.currentUser!!.usersForRoom( room.id, callback = { result ->
-            when (result) {
-                is Result.Success -> {
-                    result.value.let { members ->
-                        //check we actually have another user to talk to
-                        val otherMember = members.find { user-> user.id != ChatkitManager.currentUser!!.id }
-                        if (otherMember == null) {
-                            val error = "Couldn't find any other people in room " + room.name
-                            Log.e(ChatkitManager.LOG_TAG, error)
-                            view?.onError(error)
-                        } else {
-                            view?.onPerson(otherMember, room)
-                        }
-                    }
-                }
+    private fun reconcileUsers(users: List<User>) {
+
+        for (room in ChatkitManager.currentUser!!.rooms) {
+
+            val otherMemberId = room.memberUserIds.find { userId-> userId != ChatkitManager.currentUser!!.id }
```

Comment:
```suggestion
            val otherMemberId = room.memberUserIds.find { userId-> userId != ChatkitManager.currentUser!!.id }!!
```
to simplify the code as in this demo app a user can't leave a room. We already have an assumption that there is **one** other member so it's also good to assume that that member is there.